docs: Extend documentation on allowed-gadgets. (#3473)
inspektor-gadget/inspektor-gadget: docs/reference/restricting-gadgets.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

It is possible to limit the gadgets that can be run by using the `--allowed-gadgets` flag.
With this flag, you can restrict running the gadgets using the following:
1. By digest,
2. By tag

> 3. Or by using a prefix, like `ghcr.io/inspektor-gadget/gadget/*.

Reviewer: ```suggestion
3. Or by using a prefix with a wildcard at the end, like `ghcr.io/inspektor-gadget/gadget/*`.
```

> By default, all gadgets are allowed.

Reviewer: ```suggestion
By default, all gadgets are allowed, although other restrictions (like a signature check) could still keep it from running.
```

> By default, all gadgets are allowed, although other restrictions (like a signature check) could still keep it from running.

Reviewer 2: ```suggestion
By default, all gadgets are allowed, although other restrictions (like a signature check) could still keep them from running.
```

?

```
</TabItem>
</Tabs>

### By tag

> We also offer the posibility to restrict running by gadget tags:

Reviewer: ```suggestion
We also offer the possibility to restrict running by image tags:
```
Unsure about this one.
